### PR TITLE
EBP-64: Added a Nack sample to demonstrate how to use the Nack feature in the Go PubSub+ API

### PR DESCRIPTION
**Pull request by oodigie**
The goal of this PR is to add sample(s) to demonstrate the use of the Nack feature in the Solace PubSub+ Go API.

**Changes:**

- EBP-64: Add a Nack code sample to the Github samples repository
- EBP-64: Bumped version of the `solace/dev/go/messaging package` to `v1.8.0`  which is the expected release version with the Nack feature

### CODE REVIEW

**Comment by oodigie** [comment]
I marked this as a draft PR since the Nack feature for the PubSub+ Go API is not yet released. This is to prevent accidentally merging it before then.

File: patterns/guaranteed_receiver_nack.go
Comment (TrentDaniel) [inline patterns/guaranteed_receiver_nack.go:23]: @oodigie  Is it conventional to leave commented logging statements in the samples?

File: patterns/guaranteed_receiver_nack.go
Comment (TrentDaniel) [inline patterns/guaranteed_receiver_nack.go:64]: @oodigie , We should have a separate sample function or module showing how to configure each one instead of leaving them in comments.

File: patterns/guaranteed_receiver_nack.go
Comment (TrentDaniel) [inline patterns/guaranteed_receiver_nack.go:96]: @oodigie Why is this commented?

File: patterns/guaranteed_receiver_nack.go
Comment (TrentDaniel) [inline patterns/guaranteed_receiver_nack.go:100]: The different settlements should be used in different functions or modules for each type, so that the user actually has a sample of how to use each of the different types

File: patterns/guaranteed_receiver_nack.go
Comment (TrentDaniel) [inline patterns/guaranteed_receiver_nack.go:122]: @oodigie , you need to add a comment justifying the timeout value that you gave so that the user understands whether it was arbitrary, or if a 1 second timeout is necessary/recommended for this use case.

**Review by TrentDaniel — CHANGES_REQUESTED** [review]
The general contents are fine, but there are some required changes before we can consider this complete.

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go:23]: @TrentDaniel,
Yes, that is the normal convention in the other samples written by the DAs.

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go:96]: @TrentDaniel, this is the same way the other samples were written by the DAs.

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go]: Okay. I have made the update

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go]: Okay. I have made the update

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go:117]: No, like I stated earlier. This is the way the other samples from the DAs are organized. There are commented out lines left as cues for the user of the samples.

File: patterns/guaranteed_receiver_nack.go
Comment (oodigie) [reply patterns/guaranteed_receiver_nack.go:202]: @TrentDaniel, I don't think this is really necessary, please see the other samples. The Go PubSub+ API documentation clearly states what a grace period is and there is no explanation as to why we use 1 second or 10 seconds. The selected grace period is at the discretion of the dev.

Comment (TrentDaniel) [reply patterns/guaranteed_receiver_nack.go:202]: @oodigie , if a new customer is trying to learn about how to use our product or this feature, they could be confused as to why a specific grace period is given. They might wonder if 1 second is passively being recommended as the minimum grace period on termination for this feature, since it is not configurable through the sample and it there is no explanation. If this is done elsewhere in the samples, then those are just additional examples of ambiguity, and do not justify this instance of ambiguity. Please either state in the comment that the timeout is arbitrary, or that it was selected as 1 second for a specific reason.

**Comment by oodigie** [comment]
@TrentDaniel, again what you are suggesting is very subjective. The documentation already explains what a grace period is on Termination. If we want this reviewed, we should defer to the Developer Advocates (@TamimiGitHub, @aaron-613). You can open an issue against the samples so that one of the DAs can look into it and decide if this is worth doing across the samples. From my previous approved PRs into this samples repository, the general expectation from the DAs is that the samples should be relatable (similar in terms of organization) as the possible to the other ones.

Please take a look at the other samples.

**Comment by TrentDaniel** [comment]
> @TrentDaniel, again what you are suggesting is very subjective. The documentation already explains what a grace period is on Termination. If we want this reviewed, we should defer to the Developer Advocates (@TamimiGitHub, @aaron-613). You can open an issue against the samples so that one of the DAs can look into it and decide if this is worth doing across the samples. From my previous approved PRs into this samples repository, the general expectation from the DAs is that the samples should be relatable (similar in terms of organization) as the possible to the other ones.
> 
> Please take a look at the other samples.

@oodigie , You are responsible for the contents of the PR, if you would like to defer to one of the DAs for review, it is your responsibility to make sure that happens. I have already left my review. If one of the DAs disagrees with me and approves the PR, that's good enough for me :).

**Comment by oodigie** [comment]
@aaron-613, I converted this draft PR back into a normal PR now that the Go PubSub+ API with the Nack has been released - https://pkg.go.dev/solace.dev/go/messaging@v1.8.0.

Kindly help to review and merge this PR if everything looks okay. Thanks

**Review by TrentDaniel — APPROVED** [review]
LGTM

**Review by TrentDaniel — APPROVED** [review]
LGTM

**Pull request merged**
